glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -24,9 +24,13 @@ class GleanTest {
         val context: Context = ApplicationProvider.getApplicationContext()
         val glean = GleanInternalAPI()
         glean.initialize(context)
+
         glean.handleBackgroundEvent()
+
         val pingPath = File(context.applicationInfo.dataDir, "glean_data/pings")
         val path = pingPath.toString()
-        assertEquals(2, pingPath.listFiles()?.size)
+
+        // All pings are empty, thus none should be stored
+        assertNull(pingPath.listFiles())
```

Comment:
The `baseline` ping should not be empty: why aren't we getting one here?

Reply:
Hah, that is true, we should at least have the `glean.baseline.locale`.

Now this is what happens:

* We directly create an object of GleanInternalAPI
* Internally that initializes core metrics
	* This uses the created metrics (`GleanBaseline.locale.set(...)`)
	* `StringMetricType.set` uses `Glean.handle`, that is: the handle of the singleton (which in this case is ... `0`)

So before we change this and to still keep it testable we might need to pass/store the handle somehow into the metrics (or pass it on setting)

Reply:
Ah, I guess we need to test using the Kotlin-side `Glean` singleton here, rather than making a new one...

Reply:
That breaks down now that we have lots of parallel tests, all of them using the singleton object.

Reply:
But we aren't parallel on the Kotlin side, right?  (At least `glean-ac` isn't -- I don't know if we had to set it from the non-default).  I think it's reasonable on the Kotlin side to have tests with global state and use the `resetGlean` command to reset before each test, etc.

Reply:
You are right.
We're not parallel, just same-process, therefore re-using the (already initialized) singleton.